app/utilities/schema.py

```diff
@@ -1,37 +1,59 @@
-from app import cache
-from app.parser.v0_0_1.schema_parser import SchemaParser
-from app.schema_loader.schema_loader import load_schema
+import json
+import os
 
+from structlog import get_logger
 
-def get_schema(metadata):
-    """
-    Get the schema for the current user
-    :return: (json, schema) # Tuple of json and schema object from schema file
-    """
+from app import cache
+from app import settings
 
-    eq_id = metadata["eq_id"]
-    form_type = metadata["form_type"]
-    language_code = metadata["language_code"] if "language_code" in metadata else None
 
-    json_schema = load_and_parse_schema(eq_id, form_type, language_code)
+logger = get_logger()
 
-    return json_schema
+
+@cache.memoize()
+def load_schema_from_metadata(metadata):
+    return load_schema_from_params(metadata['eq_id'], metadata['form_type'], metadata.get('language_code'))
 
 
 @cache.memoize()
-def load_and_parse_schema(eq_id, form_type, language_code):
+def load_schema_from_params(eq_id, form_type, language_code='en'):
+    return load_schema_file("{}_{}.json".format(eq_id, form_type), language_code)
+
+
+def load_schema_file(schema_file, language_code=None):
     """
-    Use the schema loader to get the schema from disk. Then use the parse to construct the object schema
-    :param eq_id: the id of the questionnaire
-    :param form_type: the form type
-    :return: (json, schema) # Tuple of json and schema object from schema file
+    Load a schema, optionally for a specified language.
+    :param language_code: ISO 2-character code for language e.g. 'en', 'cy'
     """
-    # load the schema
+    language_code = language_code or 'en'
```

Comment:
Because calling code can pass in None; would still need this line or force callers to not pass None in. Not sure if it's better for the caller to deal with it or the callee.